bin/npm

```diff
@@ -2,12 +2,16 @@
 (set -o igncr) 2>/dev/null && set -o igncr; # cygwin encoding fix
 
 basedir=`dirname "$0"`
+isexe=0
 
 case `uname` in
     *CYGWIN*) basedir=`cygpath -w "$basedir"`;;
 esac
 
 NODE_EXE="$basedir/node.exe"
+if [ -x "$NODE_EXE" ]; then
+  isexe=1
+fi
 if ! [ -x "$NODE_EXE" ]; then
```

Comment:
maybe make this an `else` just to clarify that they're connected?  It's a bit weird to see `if (x)` followed immediately by `if (!x)`, makes me start hunting for whether `x` might've changed in the block somehow.